WPFHospitalEditor/SearchResultDialog.xaml.cs

```diff
@@ -138,6 +151,48 @@ private void addAdvancedSearchButton()
             Grid.SetColumn(advancedSearch, 4);
 
             DynamicGrid.Children.Add(advancedSearch);
+
+            advancedSearch.Click += (s, e) =>
+            {
+                if (row.ContainsKey(Grid.GetRow(advancedSearch)))
+                {
+                    row[Grid.GetRow(advancedSearch)].selected = true;
+                    mapObjectController.update(row[Grid.GetRow(advancedSearch)]);
+                    if (row[Grid.GetRow(advancedSearch)].Description.Equals(""))
+                    {
```

Comment:
In many places you use Grid.GetRow(avancedSearch)]. It is better to define the variable that you will use in all these places

Reply:
It represents choosen object to display, it's fixed now.